client/java/src/main/java/io/openlineage/client/transports/KafkaTransport.java

```diff
@@ -57,17 +57,47 @@ private String getMessageKey(@NonNull OpenLineage.RunEvent runEvent) {
     return "run:" + job.getNamespace() + "/" + job.getName() + "/" + run.getRunId().toString();
   }
 
+  private String getMessageKey(@NonNull OpenLineage.DatasetEvent datasetEvent) {
+    final OpenLineage.Dataset dataset = datasetEvent.getDataset();
+    if (dataset == null) {
+      return null;
+    }
+
+    return "dataset:" + dataset.getNamespace() + "/" + dataset.getName();
+  }
+
+  private String getMessageKey(@NonNull OpenLineage.JobEvent jobEvent) {
+    final OpenLineage.Job job = jobEvent.getJob();
+    if (job == null) {
+      return null;
+    }
+
+    return "job:" + job.getNamespace() + "/" + job.getName();
+  }
+
   @Override
   public void emit(@NonNull OpenLineage.RunEvent runEvent) {
-    final String eventAsJson = OpenLineageClientUtils.toJson(runEvent);
+    emit(OpenLineageClientUtils.toJson(runEvent), getMessageKey(runEvent));
+  }
+
+  @Override
+  public void emit(@NonNull OpenLineage.DatasetEvent datasetEvent) {
```

Comment:
I am not sure of this. Wouldn't it make more sense to define three separate topic per RunEvent, DatasetEvent and JobEvent?

Reply:
Let's move a discussion to #2616